katdal/test/test_chunkstore_s3.py

```diff
@@ -455,10 +488,11 @@ def from_url(cls, url, authenticate=True, **kwargs):
             cls.proxy_url = 'http://{}:{}'.format(proxy_host, proxy_port)
         elif url != cls.httpd.target:
             raise RuntimeError('Cannot use multiple target URLs with http proxy')
+        url = urllib.parse.urljoin(cls.proxy_url, BUCKET + '/')
         # The token only authorises the one known bucket
         token = encode_jwt({'alg': 'ES256', 'typ': 'JWT'}, {'prefix': [BUCKET]})
-        return S3ChunkStore(cls.proxy_url, timeout=TIMEOUT, retries=RETRY,
-                            token=token, **kwargs)
+        kwargs.setdefault('token', token)
+        return url, kwargs
```

Comment:
Would this function get simpler if it was written to call `super()._tweak_store_args` (passing the proxy URL if appropriate, and passing `authenticate=False`)?

Reply:
Definitely, good suggestion.